jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -304,22 +304,28 @@ private void pullBaseImages() {
   private void obtainBaseImageLayers(boolean layersRequiredLocally) {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+
     results.baseImagesAndLayers =
         executorService.submit(
             () -> {
+              ProgressEventDispatcher progressDispatcher =
```

Comment:
This is a temporary workaround to overcome the runtime failure. But I can accept this to move forward. Eventually, we should implement a permanent fix. Please keep track of this issue too.

```suggestion
              // TODO: ideally, progressDispatcher should be closed at the right moment, after the scheduled threads have completed. However, it can be tricky and cumbersome to track completion, so it may just be better to delay closing until everything ends. At least, we must ensure that it's not closed prematurely. (Garbage collection doesn't auto-close it wit the current implementation.)
              ProgressEventDispatcher progressDispatcher =
```